builtInFunctions/esdtDataStorage.go

```diff
@@ -143,7 +143,7 @@ func (e *esdtDataStorage) getESDTNFTTokenOnDestinationWithAccountsAdapterOptions
 		return nil, false, err
 	}
 
-	if !e.enableEpochsHandler.IsSaveToSystemAccountFlagEnabled() || nonce == 0 {
+	if !e.metaDataInSystemAccount(esdtData.Type) || nonce == 0 {
```

Comment:
I do not think that this is correct. As tokenType can be of NFTV2  but the token was not yet migrated. you have to check whether esdtData.MetaData is with data or not.

Reply:
Changed.